sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/HttpPipelineHelper.cs

```diff
@@ -126,5 +129,63 @@ internal static bool TryGetRetryIntervalTimespan(Response? httpResponse, out Tim
 
             return Encoding.UTF8.GetBytes(partialContent);
         }
+
+        internal static ExportResult IsSuccess(HttpMessage httpMessage)
+        {
+            if (httpMessage.HasResponse && httpMessage.Response.Status == ResponseStatusCodes.Success)
+            {
+                return ExportResult.Success;
+            }
+
+            return ExportResult.Failure;
+        }
+
+        internal static void HandleFailures(HttpMessage httpMessage, PersistentBlob blob, PersistentBlobProvider blobProvider)
+        {
+            int statusCode = 0;
+            bool shouldRetry = true;
+
+            if (httpMessage.HasResponse)
+            {
+                statusCode = httpMessage.Response.Status;
+                switch (statusCode)
+                {
+                    case ResponseStatusCodes.PartialSuccess:
+                        // Parse retry-after header
+                        // Send Failed Messages To Storage
+                        // Delete existing file
+                        TrackResponse trackResponse = GetTrackResponse(httpMessage);
+                        var content = GetPartialContentForRetry(trackResponse, httpMessage.Request.Content);
+                        if (content != null)
+                        {
+                            blob.TryDelete();
+                            blobProvider?.SaveTelemetry(content);
```

Comment:
blobProvider shouldn't be null here because it's not nullable in the method signature.